__tests__/integration-tests.test.js

```diff
@@ -138,6 +138,57 @@ describe('GET /api/articles/:article_id tests', () => {
     });
 });
 
+describe('POST /api/articles/:article_id/comments test', () => {
+    test('should respond with the comment posted with an updated comments table when given a valid article id', () => {
+        const newComment =  {
+            username:"lurker",
+            body:"I'm allergic to chicken"
+          }
+
+        return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            expect(response.body.comment).toMatchObject(
+                {
+                    body: expect.any(String),
+                    votes: expect.any(Number),
+                    author: expect.any(String),
+                    article_id: 1,
+                    created_at: expect.any(String)
+                }
+            )
+        })
+    });
+    test('should respond with a 404 when given a valid article id that does not exist', () => {
+        const newComment =  {
+            username:"lurker",
+            body:"I'm allergic to chicken"
+          } 
+        return request(app)
+        .post("/api/articles/999/comments")
+        .send(newComment)
+        .expect(404)
+        .then(({ body }) => {
+            expect(body.msg).toBe("article does not exist")
+        })
+    });
+    test('should respond with a 400 when given a invalid article id', () => {
+        const newComment =  {
+            username:"lurker",
+            body:"I'm allergic to chicken"
+          } 
+        return request(app)
+        .post("/api/articles/not-an-id/comments")
+        .send(newComment)
+        .expect(400)
+        .then(({ body }) => {
+            expect(body.msg).toBe("Bad Request")
+        })
+    });
+});
+
```

Comment:
Really good testing so far, but there are a few more tests that should be added for this endpoint.

What should happen when a user makes a request with an incomplete/incorrect request body?
What should happen when a user makes a request with a username that does not exist?